pkg/skaffold/initializer/analyze/analyze.go

```diff
@@ -106,34 +106,38 @@ func (a *ProjectAnalysis) Analyze(dir string) error {
 	// init should have the same results.
 	sort.Sort(dirents)
 
-	var subdirectories []*godirwalk.Dirent
+	var subdirectories []string
 
 	// Traverse files
 	for _, file := range dirents {
 		if util.IsHiddenFile(file.Name()) || util.IsHiddenDir(file.Name()) {
 			continue
 		}
 
+		filePath := filepath.Join(dir, file.Name())
+
 		// If we found a directory, keep track of it until we've gone through all the files first
 		if file.IsDir() {
-			subdirectories = append(subdirectories, file)
+			subdirectories = append(subdirectories, filePath)
 			continue
 		}
 
-		filePath := filepath.Join(dir, file.Name())
-		stat, err := os.Stat(filePath)
-		if err != nil {
-			// this is highly unexpected but in case there could be a racey situation where
-			// the file gets removed right between ReadDirents and Stat
-			continue
-		}
-		if a.maxFileSize > 0 && stat.Size() > a.maxFileSize {
-			logrus.Debugf("skipping %s as it is larger (%d) than max allowed size %d", filePath, stat.Size(), a.maxFileSize)
-			continue
+		if a.maxFileSize > 0 {
```

Comment:
It's initialized from the [global config's `MaxFileSize`](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/initializer/analyze/analyze.go#L88).  So it will be 0 by default.